sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,399 @@
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const T &real, const marray<T, NumElements> &imag);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<sycl::ext::oneapi::complex<T>, NumElements> &cmplx, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Get
+
+  // return marray of component
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  // return sequence of elements of component
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<T, int_seq::size()> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<T, int_seq::size()> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Set
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const marray<T, NumElements> &values);
+
+  template <class T, std::size_t NumElements>
+  void set_real(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+  template <class T, std::size_t NumElements>
+  void set_imag(marray<sycl::ext::oneapi::complex<T>, NumElements> &input, const T value);
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The class `sycl::ext::oneapi::marray<sycl::ext::oneapi::complex<T>, N>`,
+has instantiations of `T`; `float`, `double`, and `sycl::half` declared.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<double>, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<float>, NumElements>;
+
+  template <std::size_t NumElements>
+  class marray<sycl::ext::oneapi::complex<sycl::half>, NumElements>;
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The generic type `mgencomplex` is defined as types
+`marray<sycl::ext::oneapi::complex<double>, {N}>`,
+`marray<sycl::ext::oneapi::complex<float>, {N}>`,
+`marray<sycl::ext::oneapi::complex<sycl::half>, {N}>`.
```

Comment:
I think we should make it clear that the extension does not define a C++ type named `mgencomplex`.  Maybe a better way to say it is:

> The table below shows free functions operating on the `marray` type when it is specialized with `sycl::ext::oneapi::complex<double>`, `sycl::ext::oneapi::complex<float>`, and `sycl::ext::oneapi::complex<sycl::half>`.  For the purposes of this specification, we use the generic type name `mgencomplex` to represent these three specializations.  However, there is no C++ type actually named `mgencomplex`.

Reply:
I have clarified the paragraph, making it clear that there is no C++ type named `mgencomplex`